src/gt4py/backend/numpy_backend.py

```diff
@@ -188,6 +188,44 @@ def visit_TernaryOpExpr(self, node: gt_ir.TernaryOpExpr):
 
         return source
 
+    def _visit_branch_stmt(self, stmt):
+        sources = []
+        if isinstance(stmt, gt_ir.Assign):
+            condition = (
+                (
+                    "{np}.logical_and(".format(np=self.numpy_prefix)
+                    + ", ".join(
+                        [
+                            "__condition_{level}".format(level=i + 1)
+                            for i in range(self.conditions_depth)
+                        ]
+                    )
+                    + ")"
+                )
+                if self.conditions_depth > 1
+                else "__condition_1"
+            )
+
+            target = self.visit(stmt.target)
+            value = self.visit(stmt.value)
+            sources.append(
+                "{target} = vectorized_ternary_op(condition={condition}, then_expr={then_expr}, else_expr={else_expr}, dtype={np}.{dtype})".format(
+                    condition=condition,
+                    target=target,
+                    then_expr=value,  # value if is_if else target,
```

Comment:
I don't fully understand the comments here. Maybe you could improve them a bit

Reply:
I should remove them, I commented out a previous piece of code and forgot about it.